staging/src/k8s.io/kube-aggregator/pkg/apis/apiregistration/validation/validation.go

```diff
@@ -45,33 +45,42 @@ func ValidateAPIService(apiServer *discoveryapi.APIService) field.ErrorList {
 		field.NewPath("metadata"))
 
 	// in this case we allow empty group
-	if len(apiServer.Spec.Group) == 0 && apiServer.Spec.Version != "v1" {
+	if len(apiService.Spec.Group) == 0 && apiService.Spec.Version != "v1" {
 		allErrs = append(allErrs, field.Required(field.NewPath("spec", "group"), "only v1 may have an empty group and it better be legacy kube"))
 	}
-	if len(apiServer.Spec.Group) > 0 {
-		for _, errString := range utilvalidation.IsDNS1123Subdomain(apiServer.Spec.Group) {
-			allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "group"), apiServer.Spec.Group, errString))
+	if len(apiService.Spec.Group) > 0 {
+		for _, errString := range utilvalidation.IsDNS1123Subdomain(apiService.Spec.Group) {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "group"), apiService.Spec.Group, errString))
 		}
 	}
 
-	for _, errString := range utilvalidation.IsDNS1035Label(apiServer.Spec.Version) {
-		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "version"), apiServer.Spec.Version, errString))
+	for _, errString := range utilvalidation.IsDNS1035Label(apiService.Spec.Version) {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "version"), apiService.Spec.Version, errString))
 	}
 
-	if apiServer.Spec.Priority <= 0 || apiServer.Spec.Priority > 1000 {
-		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "priority"), apiServer.Spec.Priority, "priority must be positive and less than 1000"))
+	if apiService.Spec.Priority <= 0 || apiService.Spec.Priority > 1000 {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "priority"), apiService.Spec.Priority, "priority must be positive and less than 1000"))
 
 	}
 
-	if len(apiServer.Spec.Service.Namespace) == 0 {
+	if apiService.Spec.Service == nil {
+		if len(apiService.Spec.CABundle) != 0 {
```

Comment:
> These two errors checks are done both inside and outside of this if statement. Would it make more sense to move them before the Service == nil check and standardize them? (I like you new version better anyway)

I think these are different.  If you have a local APIService without a spec.service set, then you cannot have a cabundle or insecure because they wouldn't be honored.  The other check makes sure you don't set both at the same time